File: JewelsAndStones/JewelsAndStones.java
import java.util.*;
public class JewelsAndStones {
    public int numJewelsInStones(String J, String S) {
        int res = 0;
        Set<Character> m = new HashSet<Character>();
        for (char c : J.toCharArray()) {
            m.add(c);
        }
        for (char c : S.toCharArray()) {
            if (m.contains(c)) res++;
        }
        return res;
    }
    public static void main(String[] args) {
	String J = new String("aA");
	String S = new String("aAAbbb");
	JewelsAndStones s = new JewelsAndStones();
	System.out.println(s.numJewelsInStones(J, S));
    }

}

File: 771.JewelsAndStones/JewelsAndStones.cpp
#include <iostream>
#include <unordered_set>
using namespace std;

class Solution {
public:
    int numJewelsInStones(string J, string S) {
        int res = 0;
        unordered_set<char> s(J.begin(), J.end());
        for (char c : S) {
            if (s.count(c))
                res++;
        }
        return res;
    }
};

int main() {
    Solution s;
    string J = "aA", S = "aAAbbb";
    cout << s.numJewelsInStones(J, S) << endl;
    return 0;
}
